### ISSUE TITLE
[1b2c8fac] http access error: cannot connect to host:443

### ISSUE THREAD

**Issue by marinica94** [caprica/vlcj#434]
Hi, I want to skip to the next song when VLC finishes, I am getting error no 443 http access error. Does the VLC need to be resseted or something ?

                 @Override
                 public void finished(uk.co.caprica.vlcj.player.MediaPlayer mediaPlayer) {
                     if (audioList.getItems().size() > currentPosition) {
                         audioList.getSelectionModel().select(currentPosition+1);
                         List<Hit> list = null;
                         try {
                              list = api.search(audioList.getSelectionModel().getSelectedItem(), mountId, "/");
                              if (list.size() > 0){
                              mediaPlayerComponent.getMediaPlayer().playMedia(api.getDownloadURL(mountId, list.get(0).getPath()));
                                currentPosition = audioList.getSelectionModel().getSelectedIndex();
                            }
                         } catch (StorageApiException e) {
                            e.printStackTrace();
                        }
                    }
                }

Is there any diagram of the states and callbacks of VLC for java ?

**Comment by caprica** [comment]
I don't know, this is not a vlcj issue.

**Comment by sarkar4540** [comment]
The same problem is with me too. Even the same object is called from my side.

in Controller Class:
`if(!queue.isEmpty()){
                            AudioEngine.AudioElement audio2=queue.remove(0);
                            audioEngine.play(audio2);
                            playlist.setItems(new ObservableListWrapper(queue));
                        }`

In AudioEngine Class:
`public void play(AudioElement audio) {
        this.audio = audio;
        stop();
        player.playMedia(audio.getSource().toExternalForm());
        state = BUFFERING;
        stateChanged.changeDetected(state, audio);

    }`

**Comment by caprica** [comment]
I still don't see how this is a vlcj problem.

**Comment by sarkar4540** [comment]
`[00007f8c48039c58] http access error: cannot connect to r3---sn-5aap5ojx-jj0l.googlevideo.com:443`

This is the actual error showing. The music is playing fine when fired using mouse click but not working when fired using any event.

**Comment by sarkar4540** [comment]
And also `MediaPlayerAdapter.error(MediaPlayer)` is not being fired in case of this error. Rather the output is thrown into standard output stream.

**Comment by caprica** [comment]
Your description is somewhat unclear.

In any case...

The error message you post is a native error from VLC (more specifically from one of its plugins I suppose).

The vlcj error callback is not being fired because LibVLC does not send a native error event in this circumstance.

This is NOTHING at all do with vlcj, as I have already stated.

**Comment by caprica** [comment]
If you have not received a "finished" or "stopped" callback, make sure you call `mediaPlayer.stop()` and then wait for the callback. After that, play your next media. That is the only thing I can suggest.

**Comment by sarkar4540** [comment]
I am asking your help to suggest solution to this problem.
However I have some more info to provide. After the problem hasnt solved, I tried another approach. I trued to trigger the next button, which will also cause the same effect, and I knew it is working well. So I removed the above mentioned code and added:

`
                                Event.fireEvent(nextButton, new MouseEvent(MouseEvent.MOUSE_CLICKED,
                                        0,0, 0,0, MouseButton.PRIMARY, 1,
                                        true, true, true, true, true, true, true, true, true, true, null));
`

But then I realised that the same problem persists. Then I thought of using the thread of JavaFX App, hence, replaced the above with:

`
                            Platform.runLater(() -> {
                                Event.fireEvent(nextButton, new MouseEvent(MouseEvent.MOUSE_CLICKED,
                                        0,0, 0,0, MouseButton.PRIMARY, 1,
                                        true, true, true, true, true, true, true, true, true, true, null));
                            }
                            );
`

Voila! and the issue solved. But wait! Another issue has arrived. The error log states the below and the JavaFX app frame freezes.

`
SIGSEGV (0xb) at pc=0x00007fd1c555d120, pid=749, tid=0x00007fd14814b700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_151-b12) (build 1.8.0_151-b12)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.151-b12 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libpthread.so.0+0xa120]  pthread_mutex_lock+0x0
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/ani/NetBeansProjects/Dhoaa/hs_err_pid749.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
`

**Comment by sarkar4540** [comment]
Oh and also I already stopping the player before playing the next. Hope you are closer to the solution, and thanks for the awesome library!

**Comment by sarkar4540** [comment]
Okay, found a solution, using a timer:

`
sliderTimer.scheduleAtFixedRate(new TimerTask() {
                                @Override
                                public void run() {
                                    if(state==AudioEngine.UNLOADED){
                                        next();
                                        if (sliderTimer != null) {
                                            sliderTimer.cancel();
                                            sliderTimer = null;
                                        }
                                    }
                            }, 0, 1000);
`

Happy coding.. and yeah it is not  vlcj issue. Looks like threads are messing up!
